web/src/features/Dashboard/components/Dashboard.tsx

```diff
@@ -423,21 +210,98 @@ class Dashboard extends Component<Props, State> {
         err instanceof Error
           ? err.message
           : "Something went wrong, please try again.";
-      this.setState({
+      setState({
         loadingApp: false,
         gettingAppErrMsg: errorMessage,
         displayErrorModal: true,
       });
     }
   };
 
-  startFetchAppDownstreamJob = () => {
-    this.state.fetchAppDownstreamJob.start(this.fetchAppDownstream, 2000);
+  const startFetchAppDownstreamJob = () => {
+    state.fetchAppDownstreamJob.start(fetchAppDownstream, 2000);
   };
 
-  updateStatus = (): Promise<void> => {
-    const { app } = this.props;
+  const setWatchState = (newAppState: App) => {
+    setState({
+      appName: newAppState.name,
+      iconUri: newAppState.iconUri,
+      currentVersion: newAppState.downstream?.currentVersion,
+      downstream: newAppState.downstream,
+      links: newAppState.downstream?.links,
+    });
+  };
 
+  const getAppLicense = async ({ slug }: { slug: string }) => {
+    await fetch(`${process.env.API_ENDPOINT}/app/${slug}/license`, {
+      method: "GET",
+      headers: {
+        Authorization: Utilities.getToken(),
+        "Content-Type": "application/json",
+      },
+    })
+      .then(async (res) => {
+        const body = await res.json();
+        if (!res.ok) {
+          setState({ gettingAppLicenseErrMsg: body.error });
+          return;
+        }
+        if (body === null) {
+          setState({ appLicense: null, gettingAppLicenseErrMsg: "" });
+        } else if (body.success) {
+          setState({
+            appLicense: body.license,
+            gettingAppLicenseErrMsg: "",
+          });
+        } else if (body.error) {
+          setState({
+            appLicense: null,
+            gettingAppLicenseErrMsg: body.error,
+          });
+        }
+      })
+      .catch((err) => {
+        console.log(err);
+        setState({
+          gettingAppLicenseErrMsg: err
+            ? `Error while getting the license: ${err.message}`
+            : "Something went wrong, please try again.",
+        });
+      });
+  };
+
+  useEffect(() => {
+    if (props.app) {
+      setWatchState(props.app);
+      getAppLicense(props.app);
+    }
+  }, [props.app]);
```

Comment:
this replaces the `componentDidUpdate` with an effect hook that fires when the `props.app` updates